Add Cloud Composer Support for Logs in Cloud Logging Only feature (#9821)
GoogleCloudPlatform/magic-modules: mmv1/third_party/terraform/website/docs/r/composer_environment.html.markdown

  Google Compute Engine Public IPs and Google Prod IPs. Structure is
  [documented below](#nested_master_authorized_networks_config).

> * `data_retention_config` -

Reviewer: It looks like this is in the Cloud Composer v1 section - is that correct? The version of it in the v2 section says it's only supported on 2.0.23 or higher.

Author: Removed.

  (Optional)

>   The configuration setting for Airflow data retention mechanism.

Reviewer: ```suggestion
  The configuration setting for Airflow data retention mechanism. Structure is
```

Author: Done.

  Google Compute Engine Public IPs and Google Prod IPs. Structure is
  [documented below](#nested_master_authorized_networks_config).

> 

Author: Done.

* `ip_allocation_policy` -
  (Optional)
  Configuration for controlling how IPs are allocated in the GKE cluster.

>   Structure is [documented below](#nested_ip_allocation_policy). Structure is

Reviewer: ```suggestion
  Structure is [documented below](#nested_ip_allocation_policy).
```

  (Required)
  `cidr_block` must be specified in CIDR notation.

<a name="nested_data_retention_config"></a>The `data_retention_config` block supports:
* `task_logs_retention_config` - 
  (Optional)

>   The configuration setting for Task Logs.

Reviewer: ```suggestion
  The configuration setting for Task Logs. Structure is
```

Author: Done.

  [documented below](#nested_task_logs_retention_config)

<a name="nested_task_logs_retention_config"></a>The `task_logs_retention_config` block supports:
* `storage_mode` - 
  (Optional)

>   The mode of storage for Airflow workers task logs. Values for storage mode are 
>   `CLOUD_LOGGING_ONLY` for enabling logs in cloud logging only feature and 
>   `CLOUD_LOGGING_AND_CLOUD_STORAGE` for disabling logs in cloud logging only feature.

Reviewer: Generally we try to match the API documentation for features like this; I don't see these fields in the current public docs, so I can't confirm whether this matches the public docs.

However, I think this will be confusing to users as written - although this is a new "cloud logging only" feature that's getting launched, from the Terraform user perspective, this is a "storage mode" feature with two options: CLOUD_LOGGING_ONLY and CLOUD_LOGGING_AND_CLOUD_STORAGE. I could see it potentially having additional options in the future like CLOUD_STORAGE_ONLY, for example.

What would you think about something like:

```suggestion
  The mode of storage for Airflow workers task logs. Values for storage mode are 
  `CLOUD_LOGGING_ONLY` to only store logs in cloud logging and 
  `CLOUD_LOGGING_AND_CLOUD_STORAGE`  to store logs in cloud logging and cloud storage.
```
Or alternatively:

```suggestion
  The mode of storage for Airflow workers task logs. Values for storage mode are 
  `CLOUD_LOGGING_ONLY` or `CLOUD_LOGGING_AND_CLOUD_STORAGE`.
```

Author: Agreed.

  Google Compute Engine Public IPs and Google Prod IPs. Structure is
  documented below.

* `data_retention_config` -

>   (Optional, CLoud Composer 2.0.23 or newer only)

Reviewer: nit: typo

```suggestion
  (Optional, Cloud Composer 2.0.23 or newer only)
```

  Configuration setting for airflow data rentention mechanism. Structure is
  [documented below](#nested_data_retention_config).

<a name="nested_data_retention_config"></a>The `data_retention_config` block supports:
* `task_logs_retention_config` - 
  (Optional)
  The configuration setting for Task Logs. Structure is

>   [documented below](#nested_task_logs_retention_config)

Reviewer: ```suggestion
  [documented below](#nested_task_logs_retention_config).
```

  Google Compute Engine Public IPs and Google Prod IPs. Structure is
  documented below.

* `data_retention_config` -
  (Optional, CLoud Composer 2.0.23 or newer only)

>   Configuration setting for airflow data rentention mechanism.

Reviewer: ```suggestion
  Configuration setting for airflow data rentention mechanism. Structure is
  [documented below](#nested_data_retention_config).
```

Author: Done.
